docs/deploy/nginx-uwsgi.rst

```diff
@@ -0,0 +1,215 @@
+function or something like that. This "thing" that configures your application
+gives you an instance of :any:`falcon.API`.
+
+We need to expose that in some way so that uWSGI can find it, we will do that by
+creating a wsgi.py file. The contents for your file should change bases on how
+your application is configured. When all is said and done, have a variable
+called ``application`` which is your :any:`falcon.API` instance.
+
+.. code-block:: python
+  :caption: /home/myproject/src/wsgi.py
+
+  import os
+  import myproject
+
+  config = myproject.get_config(os.environ['MYPROJECT_CONFIG'])
+  application = myproject.create_api(config)
+
+This is a bare bones wsgi file. It imports your application and creates the
+:any:`falcon.API` instance that represents our application, exposing it as the
+variable called ``application``. This file is what we will point uWSGI at to
+load your application.
+
+Note that we did not call ``run`` like a developer might do within a Flask
+application during development or use `wsgiref.simple_server.make_server` , we
+simply assigned our `API` instance to ``application``. We *do not* want to start
+a server, uWSGI will do that and starting one here will have unexpected results.
+
+
+Getting Falcon behind uWSGI
+'''''''''''''''''''''''''''
+
+With our wsgi.py file setting up our application it is time to configure uWSGI.
+To do this, we create a uwsgi.ini file. In general, you shouldn't commit this
+file, it's creation should be of your deployment tool which is targeting the
+environment which are deploying.
+
+This configuration, when executed, will create a new uWSGI server which will be
+pointed at our wsgi.py file and served on 127.0.0.1:8080.
+
+.. code-block:: ini
+  :caption: /home/myproject/src/uwsgi.ini
+
+  [uwsgi]
+  master = 1
+  vacuum = true
+  socket = 127.0.0.1:8080
+  enable-threads = true
+  thunder-lock = true
+  threads = 2
+  processes = 2
+  virtualenv = /home/myproject/venv
+  wsgi-file = /home/myproject/src/wsgi.py
+  chdir = /home/myproject/src
+  uid = myproject-runner
+  gid = myproject-runner
+
+
+.. note:: Thread vs Processes vs gevent
+
+  There is a lot of questions to ask when deciding how to manage the processes
+  that actually run your Python code. Are you generally CPU bound or IO bound?
+  Is your application code thread safe? How many CPU's do you have? What system
+  are you on? Do you need an in-process cache?
+
+  You will have to do some research to understand this problem and no answers
+  can be given to your specific needs. In general uWSGI is so flexible you will
+  be able to find a solution that meets your applications needs. This
+  configuration enables both threads and processes. Reasearch, experiment, know
+  what you are doing, be smart.
+
+
+There are some important items in here, like ``uid`` and ``gid``. Note that
+those values are set to the application runner and not the owner of the
```

Comment:
Fix run-on sentence: "...not the owner of the application which should be the..."